pkg/webhook/resources/restore/validator.go

```diff
@@ -95,30 +121,92 @@ func (v *restoreValidator) Create(request *types.Request, newObj runtime.Object)
 	return nil
 }
 
-func (v *restoreValidator) checkBackupTarget(vmRestore *v1beta1.VirtualMachineRestore) error {
-	// get backup target
+func (v *restoreValidator) getVmBackup(vmRestore *v1beta1.VirtualMachineRestore) (*v1beta1.VirtualMachineBackup, error) {
+	vmBackup, err := v.vmBackup.Get(vmRestore.Spec.VirtualMachineBackupNamespace, vmRestore.Spec.VirtualMachineBackupName)
+	if err != nil {
+		return nil, fmt.Errorf("can't get vmbackup %s/%s, err: %w", vmRestore.Spec.VirtualMachineBackupNamespace, vmRestore.Spec.VirtualMachineBackupName, err)
+	}
+
+	if vmBackup.DeletionTimestamp != nil {
+		return nil, fmt.Errorf("vmbackup %s/%s is deleted", vmBackup.Namespace, vmBackup.Name)
+	}
+
+	if !ctlbackup.IsBackupReady(vmBackup) {
+		return nil, fmt.Errorf("VMBackup %s/%s is not ready", vmBackup.Namespace, vmBackup.Name)
+	}
+
+	return vmBackup, nil
+}
+
+func (v *restoreValidator) checkBackup(vmBackup *v1beta1.VirtualMachineBackup) error {
+	return v.checkBackupTarget(vmBackup)
+}
+
+func (v *restoreValidator) checkSnapshot(vmRestore *v1beta1.VirtualMachineRestore, vmBackup *v1beta1.VirtualMachineBackup) error {
+	if vmRestore.Namespace != vmBackup.Namespace {
+		return fmt.Errorf("Restore to other namespace with backup type snapshot is not supported")
+	}
+	if vmRestore.Spec.NewVM == false {
```

Comment:
Just my personal taste

```suggestion
	if !vmRestore.Spec.NewVM {
```

Reply:
Updated it. Thank you.